### ISSUE TITLE
Is there any data augmentation strategy used during training?

### ISSUE THREAD

**Issue by qiusuor** [InterDigitalInc/CompressAI#19]
I try to train ScaleHyperprior model using code provide in example/train.py with Imagenet2012/DIV2K dataset. The learning rate for main optimizer and aux_optimizer both set to 1e-4. The learning rate of the main optimizer is then divided by 2 when the evaluation loss reaches a plateau as it described in https://interdigitalinc.github.io/CompressAI/zoo.html. However, atfer nearly 1 week's training, the rate-distortion performance on Kadok-24 still has a gap compared to the result provided.

I am also training model on vimeo_test_clean from Vimeo90K after 2 days, it seems will not  to converge to the result provided.
Have I missed something? Is there any data augmentation strategy used during training?

**Comment by jbegaint** [comment]
Hi, no we don't do any augmentation besides random cropping the data to patches of size 256x256. We used the vimeo dataset from [here](https://github.com/anchen1011/toflow/blob/master/download_dataset.sh).

How large is the performance gap? 

Another learning rate strategy would be to set the lr to 1e-5 after ~100-150 epochs (depending on your dataset/batch size). We have observed similar performances.

**Comment by qiusuor** [comment]
Thanks for quick reply. 
The gap is huge when training with Lambda 0.0130. 
![image](https://user-images.githubusercontent.com/25458348/102970370-5adc4b00-4532-11eb-810c-385b4acc7ec5.png)

I will try to train on the dataset provide above. Is all data used in training/validating or just part of them?

**Comment by qiusuor** [comment]
I have checked the example/train.py and found that gradients were clipped by hyper_param --clip_max_norm(default:0.1). Does it influence train result? I wonder if gradient clip operation is involved in your training process? If it is, what the value of   hyper_param --clip_max_norm?

After declare network, I explicitly call the update func like below, is that ok?
        net = ScaleHyperprior(M,N)
        net.update(force=True)

**Comment by jbegaint** [comment]
Yes ok, this gap should not happen.
We do use all the training/test images provided by the dataset (with random crop for training, and center crop for testing).

We use gradient norm clipping, but usually the value is set 1. not 0.1, I'll fix the example. Thanks for reporting!

You only need to call `.update()` after the training of your network, it's only required for the entropy coding (`compress/decompress`).

**Comment by qiusuor** [comment]
Thanks. Finally I get similar results.

**Comment by jbegaint** [comment]
Great, thanks for the update!

**Comment by achel-x** [comment]
> Yes ok, this gap should not happen. We do use all the training/test images provided by the dataset (with random crop for training, and center crop for testing).
> 
> We use gradient norm clipping, but usually the value is set 1. not 0.1, I'll fix the example. Thanks for reporting!
> 
> You only need to call `.update()` after the training of your network, it's only required for the entropy coding (`compress/decompress`).

Hi, 
I met the same question. The training dataset is DIV2k. I down-sampled all 800 images to half of their sizes. All 1600 images are included. I just train one point, the performance is shown in the figure below
![image](https://user-images.githubusercontent.com/67953243/209552528-57f3b5e4-540a-442f-81d6-17326f65bcb9.png)

Here is a gap with the homepage shows.
The training set is follow the default set
epochs 100
lr is 1e-4
lamda is 1e-2
Here is part of training log.
![image](https://user-images.githubusercontent.com/67953243/209553397-31b60457-0dfc-4513-8947-84b71ea39ae7.png)


1、I wonder where is the exact position of the `update` call.
2、I noticed that the training set mentioned on the homepage is from Vimeo90k. I also want to know exactly how many images were used in training

**Comment by achel-x** [comment]
@jbegaint sorry, i found the update command. While I try to update the trained model as `python -m compressai.utils.update_model ./checkpoint.pth.tar` -a bmshj2018-factorized` 
Then to eval its performance. The results are the same![image](https://user-images.githubusercontent.com/67953243/209558494-2850eedf-be2a-49da-96d9-d161d245bb3e.png)
The name of model file in command does not cause this problem.
